scheduler/util.go

```diff
@@ -838,13 +838,16 @@ func genericAllocUpdateFn(ctx Context, stack Stack, evalID string) allocUpdateTy
 			Tasks: option.TaskResources,
 			Shared: structs.AllocatedSharedResources{
 				DiskMB: int64(newTG.EphemeralDisk.SizeMB),
-				// Since this is an inplace update, we should copy network
-				// information from the original alloc. This is similar to
-				// how we copy network info for task level networks above.
-				Networks: existing.AllocatedResources.Shared.Networks,
 			},
 		}
 
+		// Since this is an inplace update, we should copy network
+		// information from the original alloc. This is similar to how
+		// we copy network info for task level networks above.
```

Comment:
Add a note about how this is only nil when upgrading from older versions of Nomad? - Otherwise I could see this being accidentally refactored into a regression.

Reply:
Good call. Turned out not to be too hard to test. Test panicked against the old code; passes against the new and asserts AllocatedResources gets set!